### PR TITLE
Implement API Token Configure logic

### PR DESCRIPTION
**Pull request by kierson**

### CODE REVIEW

File: packages/neptune-extension/src/common/components/App.tsx [hunk 1]
```diff
@@ -1,114 +1,76 @@
...
+          onClose={() => setConfigureModalOpen(false)}
+        />
+      )}
+    </div>
   );
 };
 
 export default App;
 
+function validateGlobalApiToken() {
```
Comment (themithy) [inline packages/neptune-extension/src/common/components/App.tsx]: I suggest moving this to sth like `/common/hooks/auth.ts`

File: packages/neptune-extension/src/labextension/index.tsx [hunk 2]
```diff
@@ -29,7 +31,14 @@ class Extension implements DocumentRegistry.IWidgetExtension<NotebookPanel, INot
 
     context.ready.then(() => {
       const platformNotebook = new Notebook(context, this.app);
-      ReactDOM.render(<App platformNotebook={platformNotebook} />, widget.node);
+
+      const store = configureStore();
+
+      ReactDOM.render((
+        <Provider store={store}>
```
Comment (themithy) [inline packages/neptune-extension/src/labextension/index.tsx]: Please move the store logic to a separate component like e.g. `Bootstrap`. The code does not belong to extension initialization and is duplicated across extensions.

Comment (themithy) [reply packages/neptune-extension/src/labextension/index.tsx]:
```
ReactDOM.render(<Bootstrap platformNotebook={platformNotebook} />, ...)
------
Bootstrap = (props) => (
  <Provider>
    <App {...props} />
  </Provider>
)
```

File: packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+import React from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {
+  Configuration as BackendApiConfiguration,
+  DefaultApi as BackendApi
+} from 'generated/backend-client/src';
+
+import Modal from 'common/components/modal/Modal';
```
Comment (themithy) [inline packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx]: If you intend to change or tidy up project organization please factor it out to separate PR. First of all it makes the current commit bloated and second introducing such convention deserves a discussion in my opinion. What about `common/ui/Modal`, which is more domain-oriented ? It seems favourable to group files by functionality, not function signature.

File: packages/neptune-extension/src/labextension/index.tsx (hunk 2 above)
Comment (kierson) [reply packages/neptune-extension/src/labextension/index.tsx]: Agree.

File: packages/neptune-extension/src/common/components/App.tsx (hunk 1 above)
Comment (kierson) [reply packages/neptune-extension/src/common/components/App.tsx]: ok

File: packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx (hunk 3 above)
Comment (kierson) [reply packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx]: I disagree.
We already had convention of grouping `component.tsx` + `component.less` in a folder with the same name. I just fixed that in case of `Modal`.

I don't see benefits of seperate PR, but if you insist I will revert it.

File: packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      <button
+        children="Connect"
+        disabled={localApiToken === undefined || isLocalApiTokenValid === false}
+        onClick={handleConnect}
+      />
+    </Modal>
+  );
+};
+
+function validateLocalApiToken(apiToken: string | undefined, setValid: (valid: boolean) => void) {
```
[the remark this answers is not in the document]
Comment (themithy) [reply packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx]: Or even you could have a function like this:
```
function useTokenValidation (apiToken, onSuccess, onFailure) {
  // wrap it with useEffect or not.
  const apiTokenParsed = parseApiToken(apiToken)
  if (!apiToken || !apiTokenParsed) {
    onFailure()
  }
  makeTrialConnection(.....bothTokens)
     .then(onSuccess)
     .catch(onFailure)
}
```

File: packages/neptune-extension/src/common/state/configuration/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  apiTokenParsed: initialApiTokenParsed,
+};
+
+export function configurationReducer(state: ConfigurationState = initialState, action:ConfigurationActions): ConfigurationState {
+  switch (action.type) {
+    case "API_TOKEN_SET": {
+      return {
+        ...state,
+        apiToken: action.payload.token,
+        apiTokenParsed: parseApiToken(action.payload.token),
```
Comment (themithy) [inline packages/neptune-extension/src/common/state/configuration/reducer.ts]: It looks like in any case when the function is called, the token is already parsed, so there should be no need to parse it here again.

File: packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx (hunk 3 above)
Comment (themithy) [reply packages/neptune-extension/src/common/components/configure-modal/ConfigureModal.tsx]: My main concern is about the convention itself and pushing it to sth more domain-oriented. Less about whether it will be one or two commits.
